Samples/Samples.Cards/Program.cs

```diff
@@ -33,93 +52,102 @@ public static void Main(string[] args)
             {
-                log.Info("[JSON] JSON deserialization card requested");
+                context.Log.Info("[JSON] JSON deserialization card requested");
                 var card = CreateCardFromJson();
-                await client.Send(card);
+                await context.Send(card);
             }
             else if (text.Contains("reply"))
             {
-                await client.Send("Hello! How can I assist you today?");
+                await context.Send("Hello! How can I assist you today?");
             }
             else
             {
-                await client.Typing();
-                await client.Send($"You said '{activity.Text}'. Try typing: card, profile, validation, feedback, form, json, or reply");
+                await context.Typing();
+                await context.Send($"You said '{activity.Text}'. Try typing: card, profile, validation, feedback, form, json, or reply");
             }
-        }
+        });
 
-        [Microsoft.Teams.Apps.Activities.Invokes.AdaptiveCard.Action]
-        public async Task<ActionResponse> OnCardAction([Context] AdaptiveCards.ActionActivity activity, [Context] IContext.Client client, [Context] Microsoft.Teams.Common.Logging.ILogger log)
+        // Handle adaptive card actions using minimal API
+        AdaptiveCardInvokes.AppInvokeActivityExtensions.OnAdaptiveCardAction(teams, async context =>
         {
-            log.Info("[CARD_ACTION] Card action received");
+            var activity = context.Activity;
+
+            context.Log.Info("[CARD_ACTION] Card action received");
 
             var data = activity.Value?.Action?.Data;
 
             // Let's log the actual data structure to understand what we're working with
-            log.Info($"[CARD_ACTION] Raw data: {System.Text.Json.JsonSerializer.Serialize(data)}");
+            context.Log.Info($"[CARD_ACTION] Raw data: {System.Text.Json.JsonSerializer.Serialize(data)}");
 
             if (data == null)
             {
-                log.Error("[CARD_ACTION] No data in card action");
-                return new ActionResponse.Message("No data specified") { StatusCode = 400 };
+                context.Log.Error("[CARD_ACTION] No data in card action");
+                return new Microsoft.Teams.Api.AdaptiveCards.ActionResponse.Message("No data specified") { StatusCode = 400 };
             }
 
             // Extract action from the Value property
             string? action = data.TryGetValue("action", out var actionObj) ? actionObj?.ToString() : null;
 
             if (string.IsNullOrEmpty(action))
             {
-                log.Error("[CARD_ACTION] No action specified in card data");
-                return new ActionResponse.Message("No action specified") { StatusCode = 400 };
+                context.Log.Error("[CARD_ACTION] No action specified in card data");
+                return new Microsoft.Teams.Api.AdaptiveCards.ActionResponse.Message("No action specified") { StatusCode = 400 };
```

Comment:
got this error.. is this expected?

<img width="391" height="252" alt="Image" src="https://github.com/user-attachments/assets/d23d7bdd-450e-4e3e-bd09-80cff233d273" />